bindings/wasm/src/resolver/mixed_resolver.rs

```diff
@@ -0,0 +1,287 @@
+use wasm_bindgen_futures::JsFuture;
+
+use crate::common::PromiseVoid;
+use crate::credential::WasmFailFast;
+use crate::credential::WasmPresentation;
+use crate::credential::WasmPresentationValidationOptions;
+use crate::error::JsValueResult;
+use crate::error::WasmError;
+use crate::resolver::constructor_input::ResolverConfig;
+use crate::resolver::supported_document_types::OptionArraySupportedDocument;
+use crate::resolver::supported_document_types::OptionSupportedDocument;
+use crate::resolver::supported_document_types::RustSupportedDocument;
+use crate::stardust::WasmStardustDID;
+use crate::stardust::WasmStardustIdentityClient;
+
+use super::supported_document_types::PromiseArraySupportedDocument;
+use super::supported_document_types::PromiseSupportedDocument;
+use crate::error::Result;
+use crate::error::WasmResult;
+use wasm_bindgen::prelude::*;
+use wasm_bindgen::JsCast;
+use wasm_bindgen_futures::future_to_promise;
+
+/// Convenience type for resolving DID documents from different DID methods.   
+///  
+/// Also provides methods for resolving DID Documents associated with
+/// verifiable `Credentials` and `Presentations`.
+///
+/// # Configuration
+/// The resolver will only be able to resolve DID documents for methods it has been configured for in the constructor.
+#[wasm_bindgen(js_name = MixedResolver)]
+pub struct MixedResolver(Rc<SingleThreadedResolver>);
+
+#[wasm_bindgen(js_class = MixedResolver)]
+impl MixedResolver {
+  /// Constructs a new `MixedResolver`.
+  ///
+  /// # Errors
+  /// If both a `client` is given and the `handlers` map contains the "iota" key the construction process
+  /// will throw an error as it is then ambiguous what should be .
+  #[wasm_bindgen(constructor)]
+  pub fn new(config: ResolverConfig) -> Result<MixedResolver> {
+    let mut resolver = SingleThreadedResolver::new();
+
+    let mut attached_iota_method = false;
+    let resolution_handlers = config.handlers();
+    let client = config.client();
+
+    if let Some(handlers) = resolution_handlers {
+      let map: &Map = handlers.dyn_ref::<js_sys::Map>().ok_or_else(|| {
+        WasmError::new(
+          Cow::Borrowed("ResolverError::ConstructionError"),
+          Cow::Borrowed(
+            "could not construct resolver: the constructor did not receive a map of asynchronous functions",
+          ),
+        )
+      })?;
+      Self::attach_handlers(&mut resolver, map, &mut attached_iota_method)?;
+    }
+
+    if let Some(wasm_client) = client {
+      if attached_iota_method {
+        Err(WasmError::new(
+          Cow::Borrowed("ResolverError::ConstructionError"),
+          Cow::Borrowed("could not construct resolver: cannot attach the iota method twice"),
+        ))?;
+      }
+
+      let ref_counted_client = Rc::new(wasm_client);
+      let handler = move |did: CoreDID| {
+        //Take CoreDID (and convert internally to StardustDID) to be consistent with the other handlers (See
+        // `Self::attach_handler`).
```

Comment:
Not a leftover TODO, 
I wanted to explain why the closure takes an argument of type `CoreDocument` and not  `StardustDocument` directly even though that would be possible. The point is that all other handlers passed in from JS take `CoreDocument` (which handlers may convert internally into whatever they want) hence I wanted to do the same here, otherwise one may obtain inconsistent error messages depending on whether the `iota` method handler was passed as a `Client` or as part of the `Map`. 

If you can think of a better way of concisely explaining that then feel free to replace my comment :+1: